module/parser/src/jsTest/kotlin/org/cru/godtools/shared/tool/parser/internal/UsesResources.js.kt

```diff
@@ -12,8 +12,8 @@ internal actual val UsesResources.TEST_XML_PULL_PARSER_FACTORY: XmlPullParserFac
     get() = object : JsXmlPullParserFactory() {
         override fun readFile(fileName: String) = try {
             val path = resourcesDir?.let { "$RESOURCES_ROOT/$it" } ?: RESOURCES_ROOT
-            Promise.resolve(Resource("$path/$fileName").readText())
+            Promise.resolve(Resource("$path/$fileName").takeIf { it.exists() }?.readText())
```

Comment:
the resource framework I'm using to do unit tests doesn't actually distinguish between a file missing and having an error reading the file :(

So, by manually checking if the file exists instead of deferring to the framework I can distinguish between a file not existing and an error reading it.

Reply:
The signature of the `readFile()` method could probably still use some improvements instead of relying on returning null to indicate that a file wasn't found.

But I'll let those improvements be defined based on how the actual override is written and interacts with Kotlin/JS